sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline.py

```diff
@@ -2666,6 +2666,80 @@ def test_dsl_pipeline_with_data_transfer_copy_2urifolder(self, client: MLClient)
         }
         assert expected_job == actual_job
 
+    def test_output_setting_path(self, client: MLClient) -> None:
+        component_yaml = components_dir / "helloworld_component.yml"
+        component_func1 = load_component(source=component_yaml)
+
+        # case 1: only node level has setting
+        @dsl.pipeline()
+        def pipeline():
+            node1 = component_func1(component_in_number=1, component_in_path=job_input)
+            node1.outputs.component_out_path.path = "azureml://datastores/workspaceblobstore/paths/outputs/1"
+            return node1.outputs
+
+        pipeline_job = pipeline()
+        pipeline_job.settings.default_compute = "cpu-cluster"
+        pipeline_job = assert_job_cancel(pipeline_job, client)
+        job_dict = pipeline_job._to_dict()
+        expected_node_output_dict = {
+            'component_out_path': '${{parent.outputs.component_out_path}}',
+        }
+        expected_pipeline_output_dict = {
+            'component_out_path': {
+                # default mode added by mt, default type added by SDK
+                'mode': 'rw_mount', 'type': 'uri_folder',
+                # node level config will be copied to pipeline level
+                'path': 'azureml://datastores/workspaceblobstore/paths/outputs/1',
+            }
+        }
+        assert job_dict["jobs"]["node1"]["outputs"] == expected_node_output_dict
+        assert job_dict["outputs"] == expected_pipeline_output_dict
+
+    def test_pipeline_component_output_setting(self, client: MLClient) -> None:
+        component_yaml = components_dir / "helloworld_component.yml"
+        component_func1 = load_component(source=component_yaml)
+
+        @dsl.pipeline()
+        def inner_pipeline():
+            node1 = component_func1(component_in_number=1, component_in_path=job_input)
+            node1.outputs.component_out_path.path = "azureml://datastores/workspaceblobstore/paths/outputs/1"
+            # node1's output setting will be copied to pipeline's setting
+            return node1.outputs
+
+        @dsl.pipeline()
+        def outer_pipeline():
+            # inner_pipeline's output setting will be copied to node1's setting
+            node1 = inner_pipeline()
+            # node1's output setting will be copied to pipeline's setting
+            return node1.outputs
+
+        pipeline_job = outer_pipeline()
+        pipeline_job_dict = pipeline_job._to_dict()
+        assert pipeline_job_dict["outputs"] == {
+            'component_out_path': {'path': 'azureml://datastores/workspaceblobstore/paths/outputs/1'}
+        }
+        pipeline_component = pipeline_job.jobs["node1"].component
+        pipeline_component_dict = pipeline_component._to_dict()
+        assert pipeline_component_dict["outputs"] == {'component_out_path': {'type': 'uri_folder'}}
```

Comment:
maybe use a uri_file test case here, so we know this is not a default value here

Reply:
I saw there is ut covering this below